### PR TITLE
doc: add release notes for 12.2.11 luminous

### PR DESCRIPTION
**Pull request by theanalyst**
Signed-off-by: Abhishek Lekshmanan <abhishek@suse.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by theanalyst** [comment]
jenkins render docs

**Comment by theanalyst** [comment]
Added content from PendingReleaseNotes, tnx @neha-ojha  for bringing that to attention.

**Comment by theanalyst** [comment]
added a note on rgw fixes (cc @ivancich , @mattbenjamin )

File: doc/releases/luminous.rst [hunk 1]
```diff
@@ -1,3 +1,157 @@
...
+This is the eleventh bug fix release of the Luminous v12.2.x long term stable
+release series. We recommend that all users upgrade to this release. Please note
+the following precautions while upgrading.
+
+Notable Changes
+---------------
+
+A previous release, v12.2.9, introduced the PG hard-limit patches which were
+found to cause an issue in certain upgrade scenarios, and 12.2.10 release was
+expedited to revert those patches. If you already successfully to v12.2.9, then
```
Comment (smithfarm) [inline doc/releases/luminous.rst]: If you already successfully installed v12.2.9, . . .

File: doc/releases/luminous.rst [hunk 2]
```diff
@@ -1,3 +1,157 @@
...
+you can upgrade to 12.2.11 directly. 12.2.10 users need no special precaution.
```
Comment (smithfarm) [inline doc/releases/luminous.rst]: Should this entire block (above) start with an asterisk `*` like the blocks that follow it?

Comment (theanalyst) [reply doc/releases/luminous.rst]: makes sense will do

File: doc/releases/luminous.rst (hunk 1 above)
Comment (theanalyst) [reply doc/releases/luminous.rst]: tnx

**Comment by theanalyst** [comment]
jenkins render docs

File: doc/releases/luminous.rst [hunk 3]
```diff
@@ -1,3 +1,157 @@
...
+the following precautions while upgrading.
+
+Notable Changes
+---------------
+
+* A previous release, v12.2.9, introduced the PG hard-limit patches which were
+  found to cause an issue in certain upgrade scenarios, and 12.2.10 release was
+  expedited to revert those patches. If you already successfully installed
+  v12.2.9, then you can upgrade to 12.2.11 directly. 12.2.10 users need no
+  special precaution.
```
Comment (liewegas) [inline doc/releases/luminous.rst]: This is confusing.  12.2.10 need no precaution.  but 12.2.9 users can also upgrade directly.  isn't that everyone?

File: doc/releases/luminous.rst [hunk 4]
```diff
@@ -1,3 +1,157 @@
...
+* This release fixes the pg log hard limit bug (https://tracker.ceph.com/issues/36686).
+  A flag called `pglog_hardlimit` has been introduced. It is off by default.
+  This flag enables the feature that limits the length of the pg log. Users
+  should run `ceph osd set pglog_hardlimit` after completely upgrading to
```
Comment (liewegas) [inline doc/releases/luminous.rst]: "should"?  "must"?  regardless of which versions they previously installed?

File: doc/releases/luminous.rst [hunk 5]
```diff
@@ -1,3 +1,157 @@
...
+  12.2.11. Once all the OSDs have this flag set, the length of the pg log will
```
Comment (liewegas) [inline doc/releases/luminous.rst]: it's a single cluster-wide flag, so "once all the osds have this flag set" is confusing.

File: doc/releases/luminous.rst (hunk 3 above)
Comment (neha-ojha) [reply doc/releases/luminous.rst]: @theanalyst @liewegas I don't think we need to mention any special precaution. 
It is important to mention that users need to set the flag, in order to make use of this feature. Also, setting this flag will only succeed once the cluster has completely upgraded to 12.2.11. Does this make it clear?

Comment (theanalyst) [reply doc/releases/luminous.rst]: So everyone after upgrade must set this flag, regardless of wherever they're upgrading from, is that correct?

File: doc/releases/luminous.rst (hunk 5 above)
Comment (theanalyst) [reply doc/releases/luminous.rst]: Got it.

File: doc/releases/luminous.rst (hunk 3 above)
Comment (neha-ojha) [reply doc/releases/luminous.rst]: They are not required to, but they "must" only if they want to limit the length of their pg logs.

**Comment by theanalyst** [comment]
jenkins render docs

**Pull request merged**